llama-index-integrations/graph_rag/llama-index-graph-rag-cognee/llama_index/graph_rag/cognee/graph_rag.py

```diff
@@ -0,0 +1,116 @@
+import os
+import pathlib
+from typing import List
+
+import cognee
+
+from llama_index.core import Document
+
+from .base import GraphRAG
+
+
+class CogneeGraphRAG(GraphRAG):
+    def __init__(
+        self,
+        llm_api_key,
+        graph_db_provider,
+        vector_db_provider,
+        relational_db_provider,
+        db_name,
+    ) -> None:
+        cognee.config.set_llm_config({"llm_api_key": llm_api_key})
+
+        cognee.config.set_vector_db_config({"vector_db_provider": vector_db_provider})
+        cognee.config.set_relational_db_config(
+            {"db_provider": relational_db_provider, "db_name": db_name}
+        )
+
+        cognee.config.set_graph_database_provider(graph_db_provider)
+
+        data_directory_path = str(
+            pathlib.Path(
+                os.path.join(pathlib.Path(__file__).parent, ".data_storage/")
+            ).resolve()
+        )
+        cognee.config.data_root_directory(data_directory_path)
+        cognee_directory_path = str(
+            pathlib.Path(
+                os.path.join(pathlib.Path(__file__).parent, ".cognee_system/")
+            ).resolve()
+        )
+        cognee.config.system_root_directory(cognee_directory_path)
+
+    async def add(self, data, dataset_name: str):
```

Comment:
Can we add return type annotations, even its its `-> None` ?

Reply:
Yes, it's added